cursive-core/src/views/select_view.rs

```diff
@@ -965,13 +981,16 @@ impl<T: 'static + Send + Sync> View for SelectView<T> {
             printer.with_style(style, |printer| {
                 // Prepare the entire background
                 printer.print_hline((1, 0), x, " ");
-                // Draw the borders
-                printer.print((0, 0), "<");
-                printer.print((x, 0), ">");
+                // Draw the decorators
+                printer.print((0, 0), &self.decorators[0]);
+                printer.print((x + 1 - self.decorators[1].len(), 0), &self.decorators[1]);
```

Comment:
We should use `.width()` rather than `.len()` when the display size is what matters:
```suggestion
                printer.print((x + 1 - self.decorators[1].width(), 0), &self.decorators[1]);
```

Reply:
It may also make more sense to directly compute `x` differently above:

```rust
            let x = match printer.size.x.checked_sub(self.decorators[1].width()) {
                Some(x) => x,
                None => return,
            };
```

Reply:
Looking at this further, the use of `x` was already a bit buggy: we were trying to avoid redundantly printing under the decorators in `print_hline()` but only avoided one of them, and I think we were willing to overwrite a decorator if space-constrained by one character.  To clean this up a bit, I've redefined `x` to `lw`, the available label width after leaving space for the decorators.